### ISSUE TITLE
The column widths block attribute is not saved to the block's serialized output

### ISSUE THREAD

**Issue by guarani** [wordpress-mobile/gutenberg-mobile#4303]
**Describe the bug**

When a Tiled Gallery is inserted into the editor (using default Mosaic layout), images are added, and you switch to HTML mode, the `columnWidths` block attribute should have the correct values in the block's HTML.

**To Reproduce**
1. Add a Tiled Gallery block (defaults to Mosaic layout)
2. Add images
3. Switch to HTML mode via ellipsis button menu
4. Notice a lack of column widths in the JSON serialized block attributes

**Expected behavior**
The columns widths should be serialized to JSON just like the web does.

**Smartphone (please complete the following information):**
 - Device: iPhone
 - OS: 15
 - Version: Current `add/tiled-gallery-block` branch

**Additional info**

Consider the following Mosaic layout of three images:

| <img width="816" alt="Screen Shot 2021-11-26 at 13 12 35" src="https://user-images.githubusercontent.com/1898325/143608119-9a74ac8b-f2d5-4cf6-b580-63feec87e199.png"> |
| - |

When serialized, web has the correct `columnWidths` value while mobile does not:

| Web | Mobile |
| - | - |
| `<!-- wp:jetpack/tiled-gallery {"columnWidths":[["66.79670","33.20330"]] ...` | `<!-- wp:jetpack/tiled-gallery {"columns":2,"columnWidths":[[179,179]] ...` |

This is likely due to how in the `edit.native.js` function, the `columnsWidths` is calculated for the Square layout, but not for the Mosaic layout.

**Comment by guarani** [comment]
When fixing this bug, test by saving a post on mobile and opening on web to ensure compatibility.

**Comment by guarani** [comment]
I wanted to verify whether or not this issue affected the Square layout, or only affected other layouts such as the Mosaic. If the Square layout does not use `columnWidths`, it's not affected by this issue.

Using the Tiled Gallery on the web as reference, I found that the Square layout still includes a `columnWidth` attribute although its values seem to correspond to the Mosaic layout, which is unexpected 🤔.

To explain, here is what I see when using the Tiled Gallery's Square layout on the web:

|  |  |
| - | - |
| <img width="735" alt="Screen Shot 2022-01-04 at 13 59 36" src="https://user-images.githubusercontent.com/1898325/148096117-3f45f953-a275-412d-babc-ce4f17b29a25.png"> | <img width="894" alt="Screen Shot 2022-01-04 at 13 59 21" src="https://user-images.githubusercontent.com/1898325/148096137-5cd72711-4a99-4894-a504-b51195b5c639.png"> |

As shown below, these `columnWidths` values correspond to percentage column widths of the **Mosaic layout**, not the Square layout as I would expect.


|  |
| - |
| <img width="1486" alt="Screen Shot 2022-01-04 at 11 51 49" src="https://user-images.githubusercontent.com/1898325/148096772-618ad1af-f7e0-4cc4-bec3-b71a50a1db0a.png"> |

--- 

I think we can say that if `columnWidths` is not used by the square layout, this issue doesn't affect the Square layout. Even though it appears to not be used, the `columnWidths` value could still be needed for other reasons (e.g. backwards compatibility, compatibility with the web, etc). This is still to be determined.

**Comment by guarani** [comment]
> I think we can say that if `columnWidths` is not used by the square layout, this issue doesn't affect the Square layout. Even though it appears to not be used, the `columnWidths` value could still be needed for other reasons (e.g. backwards compatibility, compatibility with the web, etc). This is still to be determined.

I did more testing on the web and now I'm confident that `columnWidths` is **not** used by the Square layout. Take for example when I add multiple images to a Mosaic layout, and then switch to a Square layout and remove all images leaving an empty block. What I see is that `columnWidths` is still populated. Since it looks like a bug, I've created a bug report for the web here: https://github.com/Automattic/jetpack/issues/22225.

This issue is left open intentionally because there is still a bug on mobile which affects the Mosaic layout.
